src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        if (!operationHandlers.containsKey(operation)) {
+            throw new IllegalArgumentException("No handler found for operation: " + operation);
+        }
+        return operationHandlers.get(operation);
+    }
+}
```

Comment:
The OperationStrategyImpl should be placed in the `strategy` package to organize the codebase better. This aligns with the checklist item that advises to keep all logic in separate packages based on their purpose.